packages/common/src/TransactionUtils.ts

```diff
@@ -84,22 +87,44 @@ export const runTransaction = async ({
   try {
     transactionConfig = { ...transactionConfig, gas: Math.floor(estimatedGas * GAS_LIMIT_BUFFER) };
 
-    // ynatm doubles gasPrice every retry. Tries every minute (and increases gas price according to DOUBLE method) if tx
-    // hasn't mined. Min Gas price starts at caller's transactionConfig.gasPrice, with a max gasPrice of x6.
+    // ynatm doubles gasPrice or maxPriorityFeePerGas every retry depending if the transaction is a legacy or London.
+    // Tries every minute(and increases gas price according to DOUBLE method) if tx hasn't mined. Min Gas price starts
+    // at caller's transactionConfig.gasPrice or, with transactionConfig.maxPriorityFeePerGas a max gasPrice of x6.
     const gasPriceScalingFunction = ynatm.DOUBLES;
     const retryDelay = 60000;
-    if (!transactionConfig.gasPrice) throw new Error("No gas price provided");
-    const minGasPrice = transactionConfig.gasPrice;
-    const maxGasPrice = 2 * 3 * parseInt(minGasPrice);
-
-    const receipt = await ynatm.send({
-      sendTransactionFunction: (gasPrice: number) =>
-        transaction.send({ ...transactionConfig, gasPrice: gasPrice.toString() }),
-      minGasPrice,
-      maxGasPrice,
-      gasPriceScalingFunction,
-      delay: retryDelay,
-    });
+    if (!transactionConfig.gasPrice || (!transactionConfig.maxFeePerGas && transactionConfig.maxPriorityFeePerGas))
+      throw new Error("No gas information provided");
+
+    let receipt;
+    // Sending a legacy transaction (pre-London)
+    if (transactionConfig.gasPrice) {
```

Comment:
tiny nit: why do we prioritize legacy? Totally fine with it, just curious mainly.